proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,639 @@
+            "resolvedVersion": "4.8.0-preview3.5278"
+          },
+          {
+            "id": "Text2Xml.Lib",
+            "requestedVersion": "1.1.2",
+            "resolvedVersion": "1.1.2"
+          }
+        ],
+        "transitivePackages": [
+          {
+            "id": "Microsoft.CSharp",
+            "resolvedVersion": "4.0.1"
+          },
+          {
+            "id": "Microsoft.NETCore.Platforms",
+            "resolvedVersion": "1.1.0"
+          },
+  ...
+        ]
+      },
+      {
+        "framework": "net5.0",
+        "topLevelPackages": [ 
+          {
+            "id": "NuGet.Commands",
+            "requestedVersion": "4.8.0-preview3.5278",
+            "resolvedVersion": "4.8.0-preview3.5278"
+          },
+          {
+            "id": "Text2Xml.Lib",
+            "requestedVersion": "1.1.2",
+            "resolvedVersion": "1.1.2"
+          }
+        ],
+        "transitivePackages": [
+          {
+            "id": "Microsoft.CSharp",
+            "resolvedVersion": "4.0.1"
+          },
+          {
+            "id": "Microsoft.NETCore.Platforms",
+            "resolvedVersion": "1.1.0"
+          },
+  ...
+        ]
+      }
+    ]
+  }
+}
+```
+
+## Compatibility
+
+ We start with `version 1`, as long as we don't remove or rename then it'll be backward compatible. In case [we change version](https://stackoverflow.com/a/13945074) just add new properties, keep old ones even it's not used.
+
+## Out-of-scope
+
+* We won't support saving the machine-readable output to disk as part of this spec. The work-around is for the consumer to read from the console's stdout stream.
+* At this point, no other CLI commands (e.g. dotnet list reference) will be within scope for this feature.
+"--parsable" option needs separate spec.
+* Currently license info is not emitted from any cli command, it could be quite useful, we should consider in the future.
+
+## Rationale and alternatives
+Currently, no other `dotnet command` implemented this, this is the 1st time dotnet command implementing `json`(etc) output, so it could become example for others next time they implement.
+Please note, except "tab completion" (for dotnet) part all changes would be inside NuGet.Client repo(under NuGet.Core), and risk of introducing regression is low.(`--format text` refactoring related changes only come into my mind.), no impact on dotnet sdk.
+
+## Prior Art
+
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevent to this proposal? -->
+
+* https://github.com/NuGet/Home/blob/dotnet-audit/proposed/2021/DotNetAudit.md#dotnet-audit---json
+
+* https://github.com/NuGet/Home/wiki/%5BSpec%5D-Machine-readable-output-for-dotnet-list-package
+
+## Unresolved Questions
```

Comment:
@erdembayar what's your intention with the questions in this section?

I see "Unresolved questions" as a useful discussion starting point when creating the first draft of a design spec, but ideally should be resolved (and therefore removed from this section of the markdown) before the spec is accepted (PR is merged). However, you're going in the opposite direction with adding commits that add more questions to this list.

It does make it easier to find, since GitHub does a poor job at tracking comments and will collapse/hide them when there are too many. But these are duplications of comments already in the PR, and I don't understand what the plan is to get approval for this design spec, ready for someone to implement.

Unresolved questions that can be resolved later are ok, but unresolved questions that block the MVP mean the spec isn't ready to be accepted/merged.

Reply:
Thank you for feedback. I'm planning to have meeting on this then we can discuss what matters then start revolving/removing them from this section.

> It does make it easier to find, since GitHub does a poor job at tracking comments and will collapse/hide them when there are too many. But these are duplications of comments already in the PR, and I don't understand what the plan is to get approval for this design spec, ready for someone to implement.

Yes, tracking comments in Github is hard.  I just to have concise list of what need to be addressed for meeting. Otherwise, there're so many comments hard to grasp what is resolved and what is not, hard to navigate.

Reply:
I'll remove this section soon after having discussion on each.